src/main/java/Duke.java

```diff
@@ -1,10 +1,325 @@
+        this.storage.get(index).setDone();
+        Task task = this.storage.get(index);
+        return "Nice! I've marked this task as done: \n  ".concat(task.toString());
+    }
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
+        Task task = new Todo(description,state.T);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "deadline"
+    public String deadline(String description, String by) {
+        Task task = new Deadline(description,by,state.D);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "event"
+    public String event(String description, String at) {
+        Task task = new Event(description,at,state.E);
+        this.storage.add(task);
+        return "Got it. I've added this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-6: delete items from list
+    public String delete(int index) {
+        Task task = this.storage.get(index);
+        this.storage.remove(index);
+        return "Noted. I've removed this task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(this.storage.size() + "").concat(" tasks in the list."));
+    }
+
+    // wraps all messages between line separators
+    public String messageWrapper(String text) {
+        return SEP_LINE.concat("\n").concat(text).concat("\n").concat(SEP_LINE);
+    }
+
+    // decodes input from user and passes that argument to a response builder
+    public int decoder(String userInput) {
+        int res;
+        switch (userInput) {
+            case "bye":
+                res = 0;
+                break;
+            case "list":
+                res = 1;
+                break;
+            case "done":
+                res = 2;
+                break;
+            case "todo":
+                res = 3;
+                break;
+            case "deadline":
+                res = 4;
+                break;
+            case "event":
+                res = 5;
+                break;
+            case "delete":
+                res = 6;
+                break;
+            default:
+                res = 7;
+                break;
+        }
+        return res;
+    }
+
+    // A-Enums: Set Enums for Task type
+    protected enum state {
+        T,D,E;
+    }
+
+    // Level 3 -> A-Classes: Task class that represents all Tasks
+    private static class Task {
```

Comment:
Any reason you put `Task` as an inner class? Should it be in a separate file for more clarity, and later to be put in packages too?